app/src/main/java/com/github/se/studentconnect/ui/screen/camera/StoryCaptureScreen.kt

```
@@ -0,0 +1,115 @@
+package com.github.se.studentconnect.ui.screen.camera
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.github.se.studentconnect.ui.camera.CameraView
+
+/** Minimal story capture preview that toggles between live camera and inactive placeholder. */
+@Composable
+fun StoryCaptureScreen(
+    onBackClick: () -> Unit,
+    onCapture: (ByteArray) -> Unit,
+    modifier: Modifier = Modifier,
+    isActive: Boolean = true
+) {
+  Box(modifier = modifier.fillMaxSize().background(Color.Black).testTag("story_capture_screen")) {
+    if (isActive) {
+      CameraView(
+          modifier = Modifier.fillMaxSize(),
+          enableImageCapture = true,
+          onImageCaptured = {
+            // Capture button does nothing for now
+            // TODO: Implement story capture
+          },
+          onError = {
+            // Ignore errors for now
+          },
+          noPermission = { PermissionRequired(onBackClick = onBackClick) })
```

Missing error handling
Empty error handler with TODO comment. Implement proper error handling or at minimum log the error instead of silently ignoring failures.

---

nice catch! done